File: app/views/helpers/DocsHelper.java
package views.helpers;

import org.graylog2.restclient.lib.Version;

public enum DocsHelper {
    PAGE_WELCOME(""), // Welcome page of the documentation
    PAGE_ALERTS("streams.html#alerts"),
    PAGE_CLUSTER_STATUS_EXPLAINED("configuring_es.html#cluster-status-explained"),
    PAGE_COLLECTOR("collector.html"),
    PAGE_CONFIGURING_ES("configuring_es.html"),
    PAGE_DASHBOARDS("dashboards.html"),
    PAGE_ES_OPEN_FILE_LIMITS("configuring_es.html#open-file-limits"),
    PAGE_EXTRACTORS("extractors.html"),
    PAGE_FLEXIBLE_DATE_CONVERTER("extractors.html#the-flexible-date-converter"),
    PAGE_RADIO_ARCHITECTURE("architecture.html#highly-available-setup-with-graylog-radio"),
    PAGE_INDEX_MODEL("index_model.html"),
    PAGE_INDEXER_FAILURES("indexer_failures.html"),
    PAGE_LOAD_BALANCERS("load_balancers.html"),
    PAGE_SEARCH_QUERY_LANGUAGE("queries.html"),
    PAGE_STANDARD_DATE_CONVERTER("extractors.html#the-standard-date-converter"),
    PAGE_STREAM_PROCESSING_RUNTIME_LIMITS("streams.html#stream-processing-runtime-limits"),
    PAGE_STREAMS("streams.html");

    private static final String DOCS_URL = "http://docs.graylog.org/en/";

    private final String path;

    DocsHelper(String path) {
        this.path = path;
    }

    @Override
    public String toString() {
        final String version = Version.VERSION.getBranchName();
        final String baseUrl = DOCS_URL + version;

        return path.isEmpty() ? baseUrl : baseUrl + "/pages/" + path;
    }

    public String toLink(String title) {
        return "<a href=\"" + toString() + "\" target=\"_blank\">" + title + "</a>";
    }

}

File: javascript/src/util/DocsHelper.ts
'use strict';

import Version = require('./Version');

interface Pages {
  [page: string]: string;
}

class DocsHelper {
  PAGES: Pages = {
    ALERTS: 'streams.html#alerts',
    CLUSTER_STATUS_EXPLAINED: 'configuring_es.html#cluster-status-explained',
    COLLECTOR: 'collector.html',
    CONFIGURING_ES: 'configuring_es.html',
    DASHBOARDS: 'dashboards.html',
    ES_CLUSTER_STATUS_RED: 'configuring_es.html#cluster-status-explained',
    ES_CLUSTER_UNAVAILABLE: 'configuring_es.html#configuration',
    ES_OPEN_FILE_LIMITS: 'configuring_es.html#open-file-limits',
    EXTERNAL_DASHBOARDS: 'external_dashboards.html',
    EXTRACTORS: 'extractors.html',
    FLEXIBLE_DATE_CONVERTER: 'extractors.html#the-flexible-date-converter',
    INDEXER_FAILURES: 'indexer_failures.html',
    INDEX_MODEL: 'index_model.html',
    LOAD_BALANCERS: 'load_balancers.html',
    RADIO_ARCHITECTURE: 'architecture.html#highly-available-setup-with-graylog-radio',
    SEARCH_QUERY_LANGUAGE: 'queries.html',
    STANDARD_DATE_CONVERTER: 'extractors.html#the-standard-date-converter',
    STREAMS: 'streams.html',
    STREAM_PROCESSING_RUNTIME_LIMITS: 'streams.html#stream-processing-runtime-limits',
    USERS_ROLES: 'users_roles.html',
    WELCOME: '', // Welcome page to the documentation
  };
  DOCS_URL: string = 'http://docs.graylog.org/en/';

  toString(path: string): string {
    var baseUrl = this.DOCS_URL + Version.getMajorAndMinorVersion();
    return path === '' ? baseUrl : baseUrl + '/pages/' + path;
  }

  toLink(path: string, title: string): string {
    return '<a href=\'' + this.toString(path) + '\' target=\'_blank\'>' + title + '</a>';
  }
}

var docsHelper = new DocsHelper();

export = docsHelper;